plugin/metrics/setup.go

```diff
@@ -50,6 +51,27 @@ func setup(c *caddy.Controller) error {
 		return nil
 	})
 
+	c.OnRestart(func() error {
+		vars.PluginEnabled.Reset()
+		return nil
+	})
+
+	c.OnStartup(func() error {
+		conf := dnsserver.GetConfig(c)
+		plugins := conf.Handlers()
+		for _, h := range conf.ListenHosts {
+			addr, err := net.ResolveTCPAddr("tcp", net.JoinHostPort(h, conf.Port))
```

Comment:
Don't think you need `ResolveTCPAddr` here.

Reply:
So what should I do here then? Can you provide more hints, please?

Reply:
Done